wasm/__init__.py

```diff
@@ -5296,41 +5334,46 @@ def iterate_through_expression_and_validate_each_opcode(
 
 
 def instantiate_wasm_invoke_start(filename):
-    file_ = open(filename, "rb")
-    if not file_:
-        return "error, could not open " + filename
-    bytestar = memoryview(file_.read())
-    if not bytestar:
-        return "error, could not read " + filename
-    module = decode_module(bytestar)  # get module as abstract syntax
+    if not os.path.exists(filename):
+        raise ValidationError(f"Unable to open file: {filename}")
+
+    with open(filename, 'rb') as file_:
+        bytestar = memoryview(file_.read())
+        if not bytestar:
+            raise ValidationError(f"Error reading file: {filename}")
+        module = decode_module(bytestar)  # get module as abstract syntax
+
     if not module:
-        return "error, could not decode " + filename
+        raise ValidationError(f"Could not decode module: {filename}")
+
     store = init_store()  # do this once for each VM instance
     externvalstar = []  # imports, hopefully none
     store, moduleinst, ret = instantiate_module(store, module, externvalstar)
-    if moduleinst == "error":
-        return "error, module could not be instantiated"
     return ret
 
 
 def instantiate_wasm_invoke_func(filename, funcname, args):
-    file_ = open(filename, "rb")
-    if not file_:
-        return "error, could not open " + filename
-    bytestar = memoryview(file_.read())
-    if not bytestar:
-        return "error, could not read " + filename
-    module = decode_module(bytestar)  # get module as abstract syntax
+    if not os.path.exists(filename):
+        raise ValidationError(f"Unable to open file: {filename}")
+
+    with open(filename, 'rb') as file_:
+        bytestar = memoryview(file_.read())
+        if not bytestar:
+            raise ValidationError(f"Error reading file: {filename}")
+        module = decode_module(bytestar)  # get module as abstract syntax
```

Comment:
Looks like a solid DRY opportunity here, `instantiate_wasm_invoke_start` and this both have long, identical (I think) preambles. Maybe for a `#TODO`